highfidelity/hifi
# New Editor for UserData JSON

This PR adds a new, easy to use JSON editor to the userData section of edit.js.
![capture](https://cloud.githubusercontent.com/assets/843228/17876459/2c716e28-688f-11e6-8cef-aa1a2fa540fd.PNG)

To test:
1. Run the default scripts from this PR.
2. Open edit.js
3. Create a cube and go to its properties.
4. Userdata should be blank.  You should see two buttons: Edit as JSON and Clear 
5. Don't click any buttons.  Type the string 'test' in the userData.
6. Select a different object and then come back to the cube.
7. Its userdata should be the string 'test'.
8. Click the 'Clear User data button  and repeat step 6.
9. Its userdata should be blank.
10. Now click 'Edit as JSON'.
11. You should see this 
![1capture](https://cloud.githubusercontent.com/assets/843228/17876402/b551e318-688e-11e6-8af3-7f5597755ab1.PNG)
12. The button next to (empty object) can be used to append fields. Add a key and a value.  Then click "Save User Data"
13. Repeat step 6.
14. The userData should be the keys and values you entered in step 12.
15. Click the "Tree" dropdown and switch to "Code" mode
16. Enter some invalid values and try to switch back to tree or save.  You should see an error.
